enhancements/cluster-logging/workload-identity-auth-for-azure-monitor-logs.md

```diff
@@ -0,0 +1,295 @@
+  #[configurable(metadata(docs::examples = "11111111-2222-3333-4444-555555555555"))]
+  pub client_id: String,
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+1. Relies on [this upstream vector PR](https://github.com/vectordotdev/vector/pull/22912) to implement the Azure Log Ingestion sink utilizing the Log Ingestion API. This is a separate sink from the data collector API. We can support both APIs while transitioning.
+2. Current `master` branch of [upstream Vector](https://github.com/vectordotdev/vector), `>=v0.46.1`, as of 04/29/2025, utilizes `azure_identity@v0.21.0` which relies solely on environment variables for workload identity credentials and will not be sufficient when forwarding logs to multiple different Azure sinks. The aforementioned PR relies on `azure_identity@v0.21.0`.
+    - `azure_identity@v0.23.0` allows for setting `client_id`, `tenant_id`, etc. for authentication.
+    - [Workload Identity Credentials azure_identity@v0.23.0 SDK Ref](https://github.com/Azure/azure-sdk-for-rust/blob/azure_identity%400.23.0/sdk/identity/azure_identity/src/credentials/workload_identity_credentials.rs)
+    - `v0.23.0` can also utilize the [ChainedTokenCredential](https://github.com/Azure/azure-sdk-for-rust/blob/azure_identity%400.23.0/sdk/identity/azure_identity/src/chained_token_credential.rs) struct which provides a user-configurable `TokenCredential` authentication flow for applications.
+3. Additional fields will be required in sink configuration in upstream Vector's API. See [proposed API](#proposed-api) above.
+4. A separate output type, `AzureLogIngestion`, will be added to work alongside the existing `AzureMonitor` output. When Microsoft retires the  Data Collector API, the `AzureMonitor` sink can also be deprecated and retired.
+5. Long lived credentials will be implemented via `client secret` in the `AzureLogIngestion` output in the CLO API.
+
+#### Constraints
+
+- As of `v6.2`, CLO relies on `v0.37.0` of OpenShift Vector. OpenShift's Vector will have to be upgraded; however, the upgrade is currently blocked by the Rust version for RHEL.
+
+#### CCO Utility Secret
+
+After the creation of a managed identity using the CCO utility, the following secret is created and can be used for authentication:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: azure-test-secret
+  namespace: openshift-logging
+type: Opaque
+stringData:
+  azure_client_id: 11111111-2222-3333-4444-555555555555
+  azure_tenant_id: 11111111-2222-3333-4444-555555555555
+  azure_region: westus
+  azure_subscription_id: 11111111-2222-3333-4444-555555555555
+  azure_federated_token_file: /var/run/secrets/openshift/serviceaccount/token
+```
+- The `azure_federated_token_file` cannot be used because the CLO projects the service account token to a custom path. (e.g `/var/run/ocp-collector/serviceaccount/token`).
```

Comment:
I wonder if there is something we can do to make these the same so we can use it..

Reply:
Project the token to the path specified in the secret?